charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -50,15 +88,41 @@ def some_event_function():
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 1
+LIBPATCH = 2
 
-
-RELATION_NAME = "gateway-info"
-INTERFACE_NAME = "istio-gateway-info"
+# Default relation and interface names. If changed, consistency must be kept
+# across the provider and requirer.
+DEFAULT_RELATION_NAME = "gateway-info"
+DEFAULT_INTERFACE_NAME = "istio-gateway-info"
 
 logger = logging.getLogger(__name__)
 
 
+def relation_preflight_checks(
+    model: Model, relation_name: str, requirer: bool = False
+) -> None:
+    """Series of checks for ensuring the relation is properly established.
+    Args:
+        model (Model): the juju model as seen by the provider/requirer unit
+        relation_name (str): the name of the relation
+        requirer (bool, optional): set to True if the check is run in the requirer class
+    Raises:
+        GatewayRelationMissingError when there is no relation between two units
+        GatewayRelationDataMissingError when there is no data in the data bag
```

Comment:
I think RelationMissing cares about related applications, not units.  Also changing this to be consistent with `get_relation_data()`'s docstring
```suggestion
        GatewayRelationDataMissingError: if data is missing
        GatewayRelationMissingError: if there is no related application
```

Reply:
hold up - I think I'm wrong about the data being in units vs applications.  Maybe that language should stay as referring to a unit

Reply:
ok I think the new version of the lib is reading from unit, but the old version is reading from app.  See [this conversation](https://github.com/canonical/istio-operators/pull/237/files#r1170073824)